### ISSUE TITLE
[Bug] svg render glitches when used inside of CSS container with Safari

### ISSUE THREAD

**Issue by shadow-identity** [apache/echarts#20390]
### Version

5.5.1

### Link to Minimal Reproduction

https://codepen.io/shadow-identity/pen/rNXegYJ

### Steps to Reproduce

1. Set an HTML element to be a [CSS query container](https://developer.mozilla.org/en-US/docs/Web/CSS/container).
2. Add an element inside of the container, initialize echarts instance on it
3. Put some options to the echarts instance, set `legend: {type: "scroll"}`
4. Make the legend to not fit into provided space, so the legend is paginated.
5. Make the container to resize on window resize event or using ResizeObserver
6. Click on legend items
7. Resize container, keeping the legend in "scroll" state

### Current Behavior

- click on legend randomly removes or adds the legend (`<g>` element) from the SVG instance (only pagination buttons are always visible)
- changing echarts container size causes legend flickering and glitching
- stop changing echarts container can make the legend to stay disappeared
- legend appears on hover
- legend appears without any interaction after about 10-20 seconds
- legend may not be visible from the beginning if the echarts container is resized immediately after mount

### Expected Behavior

Legend should be always visible like on any other browser

### Environment

```markdown
- OS:MacOS 15.0
- Browser: 
  - stable Safari 18.0 (20619.1.26.31.6), 
  - Safari Technology Preview Release 203 (Safari 18.0, WebKit 20620.1.6)
- Framework: vanilla
```

**Comment by afiorito** [comment]
@shadow-identity did you ever find a workaround?

**Comment by shadow-identity** [comment]
@afiorito no, sadly. To us, it is quite a minor bug since we don't have many Safari users.

**Comment by gggritso** [comment]
This might be caused by this WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=319917
